### PR TITLE
Avail catalog unit test

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: libs/contract/src/lib/avails/tests/avails.availableTerritories.spec.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+describe('availableTerritories', () => {
+
+  it('France should stay available', () => {
+    const mandate = createMandate({
+      id: 'MandateA',
+      termIds: ['termA'],
+    });
+
+    const availsForm = new AvailsForm({ territories: [] }, ['duration']);
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/tests/avails.availableTerritories.spec.ts]: If you can, I would rather to that : 
```typescript
const availsForm = new AvailsForm({
  territories: [],
  duration: { from: new Date('06/01/2021'), to: new Date('12/01/2021') },
  medias: ['theatrical'],
  exclusive: true
}, ['duration']);
```
The benefice is that we could put the data into fixtures more easily

File: libs/contract/src/lib/avails/tests/avails.availableTerritories.spec.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    availsForm.get('duration').get('from').setValue(new Date('06/01/2021'));
+    availsForm.get('duration').get('to').setValue(new Date('12/01/2021'));
+    availsForm.get('medias').setValue(['theatrical']);
+    availsForm.get('exclusive').setValue(true);
+
+    const term = createTerm({ id: 'termA', contractId: mandate.id });
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/tests/avails.availableTerritories.spec.ts]: Same as above, put all data in the createTerm instead.

File: libs/contract/src/lib/avails/tests/bucketTerm.spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { AvailsFilter, isInTerm, isSameTerm } from "./../avails";
+import { createBucketTerm } from "@blockframes/contract/bucket/+state";
+
+describe('BucketTerm', () => {
+
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/tests/bucketTerm.spec.ts:5]: Split this describe into two: 
```typescript
describe('isSameTerm', () => {})
describe('isInTerm', () => {})
```

File: libs/contract/src/lib/avails/tests/territories.spec.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+describe('availableTerritories', () => {
+
+  it('France should stay available', () => {
+    const mandate = createMandate({
+      id: 'MandateA',
+      termIds: ['termA'],
+    });
+
+    const availsForm = new AvailsForm({ territories: [] }, ['duration']);
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/tests/territories.spec.ts:14]: I don't see the benefit of an `availsForm`. You should just need the value for the unit test, not the form itself.

**Review by GrandSchtroumpf — APPROVED** [review]

File: libs/contract/src/lib/avails/tests/territories.spec.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    term.exclusive = true;
+
+    const selected = toTerritoryMarker('france', 'MandateA', [mandate], term);
+    const available = availableTerritories([selected], [], [], availsForm.value, [mandate], [term]);
+
+    expect(available.length).toBe(0);
+  });
+
+  it('Return sold terms according to availsForm values', () => {
+
```
Comment (xmano) [inline libs/contract/src/lib/avails/tests/territories.spec.ts:62]: Remove extra line here

File: libs/contract/src/lib/avails/tests/territories.spec.ts [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    termD.territories = ['france'];
+    termD.exclusive = false;
+
+    const soldTerms = getSoldTerms(availsForm.value, [termA, termB, termC, termD]);
+
+    expect(soldTerms.length).toBe(0);
+  });
+
+  it('Should return available terms mandate', () => {
+
```
Comment (xmano) [inline libs/contract/src/lib/avails/tests/territories.spec.ts:146]: Extra line here

**Review by xmano — APPROVED** [review]
Approved with some comments.

And if the Date might conflict in future, you may consider deriving it from current day.

**Pull request merged**